controllers/pkg/reconcilers/bootstrap/reconciler.go

```diff
@@ -0,0 +1,277 @@
+					if ok {
+						found = true
+						clusterClient, ready, err := clusterClient.GetClusterClient(ctx)
+						if err != nil {
+							msg := "cannot get clusterClient"
+							r.l.Error(err, msg)
+							return ctrl.Result{RequeueAfter: 30 * time.Second}, errors.Wrap(err, msg)
+						}
+						if !ready {
+							r.l.Info("cluster not ready")
+							return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+						}
+						ns := &corev1.Namespace{}
+						if err = clusterClient.Get(ctx, types.NamespacedName{Name: cr.GetNamespace()}, ns); err != nil {
+							if resource.IgnoreNotFound(err) != nil {
+								msg := fmt.Sprintf("cannot get namespace: %s", secret.GetNamespace())
+								r.l.Error(err, msg)
+								return ctrl.Result{RequeueAfter: 30 * time.Second}, errors.Wrap(err, msg)
+							}
+							msg := fmt.Sprintf("namespace: %s, does not exist, retry...", cr.GetNamespace())
+							r.l.Info(msg)
+							return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+						}
+
+						if err := clusterClient.Apply(ctx, cr); err != nil {
+							msg := fmt.Sprintf("cannot apply secret to cluster %s", clusterName)
+							r.l.Error(err, msg)
+							return ctrl.Result{RequeueAfter: 10 * time.Second}, errors.Wrap(err, msg)
+						}
+					}
+				}
+			}
+			if !found {
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+		}
+	} else {
+		// this branch handles manifest installation
+		cl, ok := cluster.Cluster{Client: r.Client}.GetClusterClient(cr)
+		if ok {
+			r.l.Info("reconcile")
+			clusterClient, ready, err := cl.GetClusterClient(ctx)
+			if err != nil {
+				msg := "cannot get clusterClient"
+				r.l.Error(err, msg)
+				return ctrl.Result{RequeueAfter: 30 * time.Second}, errors.Wrap(err, msg)
+			}
+			if !ready {
+				r.l.Info("cluster not ready")
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
+			}
+
+			// install resources
+			resources, err := r.getResources(ctx, cl.GetClusterName())
+			if err != nil {
+				msg := "cannot get resources"
+				r.l.Error(err, msg)
+				return ctrl.Result{RequeueAfter: 10 * time.Second}, errors.Wrap(err, msg)
+			}
+			for _, resource := range resources {
+				r.l.Info("install manifest", "resources",
+					fmt.Sprintf("%s.%s.%s", resource.GetAPIVersion(), resource.GetKind(), resource.GetName()))
+				if err := clusterClient.Apply(ctx, &resource); err != nil {
+					r.l.Error(err, "cannot apply resource to cluster", "name", resource.GetName())
+				}
+			}
+		}
+	}
+	return ctrl.Result{}, nil
+}
+
+func (r *Reconciler) getResources(ctx context.Context, clusterName string) ([]unstructured.Unstructured, error) {
+	repos := &porchconfigv1alpha1.RepositoryList{}
+	if err := r.porchClient.List(ctx, repos); err != nil {
+		return nil, err
+	}
+
+	stagingRepoName := ""
+	for _, repo := range repos.Items {
+		if _, ok := repo.Annotations["nephio.org/staging"]; ok {
+			stagingRepoName = repo.GetName()
```

Comment:
We could end up with multiple, and then with non-deterministic behavior, depending which one gets picked (is last in the list). We should either accumulate a list here, and apply everything from all, or we should use a label on the Secret to point to a specific repo, or we should just pass a specific repo name into the controller at initialization time.

Reply:
I think it's OK, yeah.